src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -197,14 +197,14 @@ public void CheckForBrokerAccountInfoInAccountManager(IDictionary<string, string
 
                     if (string.Equals((accountData[BrokerResponseConst.UserName]).ToString(), username, StringComparison.OrdinalIgnoreCase))
                     {
-                        brokerPayload[BrokerParameter.HomeAccountId] = accountData[BrokerResponseConst.HomeAccountId].ToString();
-                        brokerPayload[BrokerParameter.LocalAccountId] = accountData[BrokerResponseConst.LocalAccountId].ToString();
+                        brokerPayload[BrokerParameter.HomeAccountId] = accountData[BrokerResponseConst.HomeAccountId]?.ToString();
```

Comment:
instead of pulling these out twice, can you store the values in a local variable?

Reply:
They are not pulled out twice, only once. The first if statement has a return at the end so after the values are pulled out the first time, the second if statement will never happen.